src/site/pages/digital/src/containers/SelectionPopup/modules/ColorModule.tsx

```diff
@@ -2,14 +2,32 @@ import {GroupAction} from "core/actions/GroupAction";
 import {ColorChangeAction} from "digital/actions/ColorChangeAction";
 import {Label, LED} from "digital/models/ioobjects";
 import {Wire} from "core/models/Wire";
+import {Node, isNode} from "core/models/Node";
+import {IOObject, Port} from "core/models";
 import {CreateModule, ModuleConfig, PopupModule} from "shared/containers/SelectionPopup/modules/Module";
+import { UndoHandler } from "core/tools/handlers/UndoHandler";
 
 
 const Config: ModuleConfig<[LED, Label, Wire], string> = {
     types: [LED, Label, Wire],
     valType: "string",
-    getProps: (o) => o.getColor(),
-    getAction: (s, newCol) => new GroupAction(s.map(o => new ColorChangeAction(o, newCol)))
+    
+    isActive: (selections) => {
+        
+        // make sure that this is only active if all the types are LED/Label/Wire/Node, but not ONLY Nodes
+        if(selections.all((s) => (isNode(s) || s instanceof LED || s instanceof Label || s instanceof Wire )) &&
+           !selections.all((s) => isNode(s))){
+          return true;
+          }
+        else{
+            return false;
+        }
```

Comment:
this can be simplified to simply
```ts
return (selections.all((s) => (isNode(s) || s instanceof LED || s instanceof Label || s instanceof Wire )) &&
        !selections.all((s) => isNode(s)));
```